github/gen-marker-methods.go

```diff
@@ -0,0 +1,256 @@
+			if !ok {
+				continue
+			}
+
+			if !mi.structFilter(ts, st) {
+				continue
+			}
+
+			t.Getters = append(t.Getters, newGetter(ts.Name.String(), mi.interfaceName, mi.methodName))
+		}
+	}
+	return nil
+}
+
+func sourceFilter(fi os.FileInfo) bool {
+	return !strings.HasSuffix(fi.Name(), "_test.go") && !strings.HasSuffix(fi.Name(), fileSuffix)
+}
+
+func (t *templateData) dump() error {
+	if len(t.Getters) == 0 {
+		logf("No getters for %v; skipping.", t.filename)
+		return nil
+	}
+
+	// Sort getters by ReceiverType.FieldName.
+	sort.Sort(byName(t.Getters))
+
+	processTemplate := func(tmpl *template.Template, filename string) error {
+		var buf bytes.Buffer
+		if err := tmpl.Execute(&buf, t); err != nil {
+			return err
+		}
+		clean, err := format.Source(buf.Bytes())
+		if err != nil {
+			return fmt.Errorf("format.Source:\n%v\n%v", buf.String(), err)
+		}
+
+		logf("Writing %v...", filename)
+		if err := os.Chmod(filename, 0644); err != nil {
+			return fmt.Errorf("os.Chmod(%q, 0644): %v", filename, err)
+		}
+
+		if err := ioutil.WriteFile(filename, clean, 0444); err != nil {
+			return err
+		}
+
+		if err := os.Chmod(filename, 0444); err != nil {
+			return fmt.Errorf("os.Chmod(%q, 0444): %v", filename, err)
+		}
+
+		return nil
+	}
+
+	if err := processTemplate(sourceTmpl, t.filename); err != nil {
+		return err
+	}
+	return processTemplate(testTmpl, strings.ReplaceAll(t.filename, ".go", "_test.go"))
+}
+
+func newGetter(receiverType, interfaceName, methodName string) *getter {
+	return &getter{
+		sortVal:       strings.ToLower(receiverType) + "." + strings.ToLower(methodName),
+		ReceiverVar:   strings.ToLower(receiverType[:1]),
+		ReceiverType:  receiverType,
+		InterfaceName: interfaceName,
+		MethodName:    methodName,
+	}
+}
+
+type templateData struct {
+	filename string
+	Year     int
+	Package  string
+	Imports  map[string]string
+	Getters  []*getter
+}
+
+type getter struct {
+	sortVal       string // Lower-case version of "ReceiverType.FieldName".
+	ReceiverVar   string // The one-letter variable name to match the ReceiverType.
+	ReceiverType  string
+	InterfaceName string
+	MethodName    string
+}
+
+type byName []*getter
+
+func (b byName) Len() int           { return len(b) }
+func (b byName) Less(i, j int) bool { return b[i].sortVal < b[j].sortVal }
+func (b byName) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+
+const source = `// Copyright {{.Year}} The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Code generated by gen-marker-methods; DO NOT EDIT.
+// Instead, please run "go generate ./..." as described here:
+// https://github.com/google/go-github/blob/master/CONTRIBUTING.md#submitting-a-patch
+
+package {{.Package}}
+{{with .Imports}}
+import (
+  {{- range . -}}
+  "{{.}}"
+  {{end -}}
+)
+{{end}}
+{{range .Getters}}
+// {{.MethodName}} implements the {{.InterfaceName}} marker interface.
```

Comment:
I don't think we need this comment for every line, do we?
Can we try deleting this comment altogether, and in fact can we also try getting rid of the blank lines between funcs?